src/components/chip/index.tsx

```diff
@@ -0,0 +1,345 @@
+import _ from 'lodash';
+import React from 'react';
+import {StyleSheet, StyleProp, ViewStyle, ViewProps, TouchableOpacityProps, ImageStyle, TextStyle, ImageSourcePropType} from 'react-native';
+import {AvatarProps, BadgeProps} from 'typings';
+// @ts-ignore
+import Assets from '../../assets';
+import {asBaseComponent} from '../../commons/new';
+import {BorderRadiuses, Colors, Spacings} from '../../style';
+// @ts-ignore
+import Avatar from '../avatar';
+// @ts-ignore
+import Badge, {BADGE_SIZES} from '../badge';
+import Image from '../image';
+import Text from '../text';
+import TouchableOpacity from '../touchableOpacity';
+import View from '../view';
+
+interface ChipProps {
+  /**
+   * Chip's size. Number or a width and height object.
+   */
+  size?: number | object;
+  /**
+   * On Chip press callback
+   */
+  onPress?: (props: any) => void;
+  /**
+   * Chip's background color
+   */
+  backgroundColor?: string;
+  /**
+   * Sets size to use minWidth and minHeight - default is true
+   */
+  minSize?: boolean;
```

Comment:
The name sounds a little like this prop act exactly as the `size` prop (meaning it accept number or an object) 
which is not such a bad idea. Either the user pass `width` or `minWidth` 

But if this prop is a boolean for another prop, I think the name should be a little different, maybe `useSizeAsMinimum` or something like that, I know it's a little long, but I guess it's clearer. 
WDYT?